### ISSUE TITLE
dynamo_export有一些缺陷

### ISSUE THREAD

**Issue by bitzyz** [bitzyz/InfiniTensor_v2.0#9]
class MatmulModel(torch.nn.Module):
    def forward(self, x, y):
        return torch.matmul(x, y)
x = torch.randn(5, 4)
y = torch.randn(4, 5)

gm, guards = torch._dynamo.export(model, x, y)
维度会被绑死
x.shape = (s77, s27)
y.shape = (s27, s77)
可见dynamo只是一次根据当前输入的trace，无法做到泛化性
需替换成torch.export
ep = torch.export.export(
    model,
    (torch.randn(5, 4), torch.randn(4, 5)),
    dynamic_shapes={
        "x": {0: Dim("N"), 1: Dim("M")},
        "y": {0: Dim("M"), 1: Dim("K")},
    },
)
